topologies/binding/new.go

```diff
@@ -114,3 +127,41 @@ func staticBinding(bindingFile string) (binding.Binding, error) {
 		pushConfig: *pushConfig,
 	}, nil
 }
+
+// rundataBind wraps an Ondatra binding to report rundata.
+type rundataBind struct {
+	binding.Binding
+}
+
+func (b *rundataBind) Reserve(ctx context.Context, tb *opb.Testbed, runTime, waitTime time.Duration, partial map[string]string) (*binding.Reservation, error) {
+	resv, err := b.Binding.Reserve(ctx, tb, runTime, waitTime, partial)
+	if err != nil {
+		return nil, err
+	}
+	b.afterReserve(ctx, resv)
+	return resv, nil
+}
+
+func (b *rundataBind) FetchReservation(ctx context.Context, id string) (*binding.Reservation, error) {
+	resv, err := b.Binding.FetchReservation(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	b.afterReserve(ctx, resv)
+	return resv, nil
+}
+
+func (b *rundataBind) afterReserve(ctx context.Context, resv *binding.Reservation) {
```

Comment:
This name is still fine, but perhaps rename it to addResvProperties given its more limited purpose now.